bundles/org.openhab.binding.haassohnpelletoven/src/main/java/org/openhab/binding/haassohnpelletoven/validation/PinValidator.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.haassohnpelletoven.validation;
+
+import java.util.regex.Pattern;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * The {@link PinValidator} is responsible for validating a given PIN.
+ *
+ * @author Christian Feininger - Initial contribution
+ */
+@NonNullByDefault
+public class PinValidator {
+
+    Pattern pattern = Pattern.compile("(\\d{4})");
```

Comment:
You could specify the pattern in the XML file. I made a suggestion below.

Reply:
ok